src/engine/fileOps/tfm.cpp

```diff
@@ -511,6 +516,29 @@ void TFMParsePattern(struct TFMParsePatternInfo info) {
       }
     }
   }
+
+  if (lastPatSeen>1) {
+    // clone the last pattern
+    info.maxPat++;
+    for (int i=0;i<6;i++) {
+      int lastPatNum = info.ds->subsong[0]->orders.ord[i][info.ds->subsong[0]->ordersLen - 1];
+      DivPattern* newPat = new DivPattern;
```

Comment:
Don't. Use getPattern instead (e.g. `DivPattern* newPat=info.ds->subsong[0]->pat[i].getPattern(info.maxPat,true)`).

Reply:
Is this a new function? The import code still directly allocates and moves new patterns above

Reply:
Nope. It's how we handle patterns in the rest of Furnace code.

getPattern() will allocate a pattern if necessary when the second argument is true.